### PR TITLE
[TR] PIM-3618: Quick export is done in backend thanks to reader/processor/writer

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                    | A
| -------------------- | ---
| Bug fix?             | n
| New feature?         | Oh yeah !
| BC breaks?           | yes
| CI currently passes? | yes
| Tests pass?          | yes
| Scenarios pass?      | yes
| Checkstyle issues?*  | n
| PMD issues?**        | n
| Changelog updated?   | YES
| Fixed tickets        | PIM-3618
| DB schema updated?   | n
| Migration script?    | n
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/DashboardBundle/Resources/public/js/last-operations-widget.js [hunk 1]
```diff
@@ -83,16 +83,20 @@ define(
...
-                    route = Routing.generate(
-                        'pim_enrich_job_tracker_show',
-                        { id: $(e.currentTarget).data('id') }
-                    );
-                } else {
-                    route = Routing.generate(
-                        'pim_importexport_' + operationType + '_execution_show',
-                        { id: $(e.currentTarget).data('id') }
-                    );
+                switch (operationType) {
```
Comment (solivier) [inline src/Pim/Bundle/DashboardBundle/Resources/public/js/last-operations-widget.js]: :+1:

File: src/Pim/Bundle/DataGridBundle/Controller/ExportController.php [hunk 2]
```diff
@@ -26,7 +26,7 @@ class ExportController
     /** @var SerializerInterface $serializer */
     protected $serializer;
 
-    /** @var ExportMassAction $exportMassAction */
+    /** @var \Pim\Bundle\DataGridBundle\Extension\MassAction\Actions\Export\ExportMassAction $exportMassAction */
```
Comment (solivier) [inline src/Pim/Bundle/DataGridBundle/Controller/ExportController.php]: why not adding this by a use like the previous one ?

File: src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php [hunk 3]
```diff
@@ -2,102 +2,109 @@
...
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\SecurityContextInterface;
 
 /**
- * Override ExportController for product exports
+ * Products quick export
  *
- * @author    Romain Monceau <romain@akeneo.com>
- * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
```
Comment (solivier) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]: You should not change the owner of the class

File: src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php [hunk 4]
```diff
@@ -2,102 +2,109 @@
...
-            $results = $this->productRepository->getFullProducts($productsList, $attributeIds);
-            echo $this->serializer->serialize($results, $this->getFormat(), $context);
-            $offset += $batchSize;
-            flush();
-            $this->productRepository->getObjectManager()->clear();
+        if (null === $token = $this->securityContext->getToken()) {
+            return null;
         }
+
+        if (!is_object($user = $token->getUser())) {
```
Comment (solivier) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]: maybe you can concatenate this if with the previous one and avoid to have to return null in the same method

File: src/Pim/Bundle/DataGridBundle/Controller/ExportController.php (hunk 2 above)
Comment (willy-ahva) [inline src/Pim/Bundle/DataGridBundle/Controller/ExportController.php]: It's not used in type hint or somewhere else. That how we do if it's only for doc.

File: src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php (hunk 3 above)
Comment (willy-ahva) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]: I totally removed the class. That's a bug from github. There is nothing from the old file

Comment (solivier) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]: The class already exists so you should keep the author

Comment (willy-ahva) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]: But there is absolutely no code from Romain here. Just the name. What is the interest to keep the name ?

File: src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php [hunk 5]
```diff
@@ -2,102 +2,102 @@
...
-            'products_export_%s_%s_%s.%s',
-            $this->catalogContext->getLocaleCode(),
-            $this->catalogContext->getScopeCode(),
-            $dateTime->format('Y-m-d_H-i-s'),
-            $this->getFormat()
+        $jobInstance = $this->jobInstanceRepository->findOneBy(['code' => 'csv_product_quick_export']);
+        $filters     = $this->gridFilterAdapter->adapt($this->request);
+        $mainContext = $filters['mainContext'];
+        unset($filters['mainContext']);
+        $rawConfiguration =
```
Comment (grena) [inline src/Pim/Bundle/DataGridBundle/Controller/ProductExportController.php]:
```php
        $rawConfiguration = addslashes(
            json_encode(
                ['filters' => $filters, 'mainContext' => $mainContext]
            )
        );
```

File: src/Pim/Bundle/EnrichBundle/Processor/MassEdit/ProductToFlatArrayProcessor.php [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+     *
+     * @param ProductInterface $product
+     *
+     * @return \Pim\Bundle\CatalogBundle\Model\ProductMediaInterface[]
+     */
+    protected function getProductMedias(ProductInterface $product)
+    {
+        $media = [];
+        foreach ($product->getValues() as $value) {
+            if (in_array(
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Processor/MassEdit/ProductToFlatArrayProcessor.php]: It's ok on a single line ;)

**Comment by grena** [comment]
GTM

**Pull request merged**